src/nodedef.h

```diff
@@ -354,7 +354,7 @@ struct ContentFeatures
 	// Player can build on these
 	bool buildable_to;
 	// Player cannot build to these (placement prediction disabled)
-	bool rightclickable;
+	bool can_place_onto;
```

Comment:
Another instance of the ironical nodedef property :)

> ```cpp
> // Player cannot build to these (placement prediction disabled)
> bool can_place_onto;
> ```

Note that this change also breaks existing mods that define nodes with the `rightclickable` property, and it's probably best to revert all changes to this nodedef property. Thoughts?

Reply:
Nodes can't be defined with the `rightclickable` property. This bool just says whether an `on_rightclick` field is set:
https://github.com/minetest/minetest/blob/4d7f2967172c5a358811ac1f24115b281eff2431/src/script/common/c_content.cpp#L507-L508

Reply:
Oh, that clears things up, thanks.

Then this might just be fixed by inverting the variable. i.e.

```cpp
lua_getfield(L, index, "on_rightclick"); 
f.can_place_onto = !lua_isfunction(L, -1); 
```

Because, if a node has an `on_rightclick` function, it is **not** placeable onto (this can be bypassed by using Shift+RMB, but that's a different story).

Reply:
There might be other factors which disallow placement without sneaking, eg. formspec meta.
Furthermore, this field has to be sent to the client, too. Because of backwards compatibility you could not send the new bool to the client without inverting it. This will cause the client code to be different from the server code (at least the names).
All in all, it's much simpler to just use keep this as is, a bool field that tells the client whether there's an `on_rightclick` callback defined.

Reply:
Hm, how about changing the name to `is_not_placeable` to not require inversion?

Reply:
(The node itself is placeable, so not `is_not_placeable` but `is_not_placeable_to`.)
The name is a bit confusing as you _can_ still place to it. You just have to hold sneak.

Reply:
I'll revert back to `rightclickable` for now.